### PR TITLE
Deprecate old test script commands

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

These aliases have been replaced with the easier to use arguments like `--debug` and `--prod` and the aliases have been proxing to the new commands for awhile now.

Deprecating the aliases now to give people a chance to use the new commands directly before removing them later.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,21 +110,21 @@
...
+    "test-persistent": "yarn test --deprecated 'yarn test --persistent'",
+    "debug-test-persistent": "yarn test --deprecated 'yarn test --debug --persistent'",
+    "test-prod": "yarn test --deprecated 'yarn test --prod'",
+    "debug-test-prod": "yarn test --deprecated 'yarn test --debug --prod'",
+    "test-prod-build": "yarn test --deprecated 'yarn test --prod --build'",
+    "test-build": "yarn test --deprecated 'yarn test --build'",
+    "test-build-prod": "yarn test --deprecated 'yarn test --build --prod'",
     "test-build-devtools": "yarn test --build --project devtools",
-    "debug-test-build-devtools": "yarn test --debug --build --project devtools",
+    "debug-test-build-devtools": "yarn test --deprecated 'yarn test-build-devtools --debug'",
```
Comment (rickhanlonii) [inline package.json:127]: In the end we'll have three base test commands:

- test
- test-www
- test-build-devtools

Comment (rickhanlonii) [reply package.json:127]: Well, we _already_ only have three commands since the aliases proxy to them.

**Review by gaearon — APPROVED** [review]
Do we reference the old commands anywhere in the docs or PR template?

**Comment by rickhanlonii** [comment]
@gaearon good thinking, updated the reference in this repo and I'll follow up with the changes to the website [here](https://reactjs.org/docs/how-to-contribute.html).

**Comment by sebmarkbage** [comment]
Please update the CI command names. They should represent the command you have to run to repro the test failure. Although not sure how you’d tell someone to put a double dash.

**Review by sebmarkbage — COMMENTED** [review]
Tbh this is a lot less discoverable. I always go to package JSON files in projects to find all the commands I might need to run.

Where is a single list of commands I need to run now?

**Comment by rickhanlonii** [comment]
Note that the only thing that's really changing between the aliases and the new commands is that --`debug`, `--prod`, `--build` and `--variant` are now options for each command instead of listing a subset of all permutations of the commands.

So instead of `test`, `test-build`, `test-prod`, and `test-build-prod` it's `test [--build] [--prod]`. The goal is to makes it easier to do what you want without searching through the aliases to find the right command. If you want to run the prod tests you add `--prod`, if you want to run the tests against the build, you add `--build`. 

This also means you can easily test missing permutations. For example, we don't have a `debug-test-prod-build`.

For a list of all supported options, we have the help message:

Screen Shot 2020-09-23 at 12 10 11 PM

Does this help @sebmarkbage?

**Comment by rickhanlonii** [comment]
Update: I added `test-stable` and `test-www-classic` so we now have a test script per release channel:

- `yarn test` (experimental)
- `yarn test-stable` (stable)
- `yarn test-www` (www-modern)
- `yarn test-classic` (www-classic)

I've also updated the circle test names to reflect the command that you would run to repro. I think we should give this a try and if we hate it we can re-evaluate.

**Pull request merged**